File: .understand-anything/.trash-1781812164/tmp/build-arch.js
const fs = require('fs');
const data = JSON.parse(fs.readFileSync('C:/Users/tahaa/Archon/.understand-anything/intermediate/assembled-graph.json', 'utf8'));
const nodes = data.nodes || [];

console.log('Total nodes to assign:', nodes.length);

const layers = [
  {
    id: "layer-infrastructure",
    label: "Infrastructure & CI/CD",
    description: "GitHub Actions workflows for CI, PyPI publishing, and GitHub Pages deployment — the build and release backbone for the Archon framework.",
    color: "#6b7280",
    order: 0
  },
  {
    id: "layer-config",
    label: "Configuration & Schemas",
    description: "Project manifest (archon.yaml), pyproject.toml, JSON/YAML schemas for all manifest types, and per-component config files that define the structure of skills, agents, bundles, pipelines, and synapses.",
    color: "#8b5cf6",
    order: 1
  },
  {
    id: "layer-component-library",
    label: "Skill & Agent Component Library",
    description: "The 99 skill directories, 14 agent definitions, 16 bundles, 5 synapses, and 9 pipelines — the declarative YAML manifests and supporting resource files that make up Archon's installable cognitive components.",
    color: "#0ea5e9",
    order: 2
  },
  {
    id: "layer-core-engine",
    label: "Core Engine",
    description: "The Python package (src/archon/) containing the CLI, registry, pipeline engine, synapse engine, policy engine, MCP schema generator, installer, and all supporting commands — the runtime brain of the Archon framework.",
    color: "#f59e0b",
    order: 3
  },
  {
    id: "layer-runtime-hooks",
    label: "Runtime Hook Layer",
    description: "Claude Code hook scripts (hooks/claude/) that fire at session start, prompt submission, pre/post tool use, and stop events to inject context, enforce guards, and track quality signals in real time.",
    color: "#ef4444",
    order: 4
  },
  {
    id: "layer-sdk-servers",
    label: "SDK & MCP Servers",
    description: "The public Python SDK (sdk/archon.py), the Rust file-ops server, the Forge MCP server, agent-router and skill-router servers — the programmatic interfaces and language-bridge components for integrating Archon into external systems.",
    color: "#10b981",
    order: 5
  },
  {
    id: "layer-tooling",
    label: "Developer Tooling & Scripts",
    description: "The VS Code extension, utility scripts (build_docs, generate-agent-cards, skill-compliance-check, etc.), test suites, and the .claude session-state XML files — tools that support development, validation, and debugging of Archon itself.",
    color: "#f97316",
    order: 6
  },
  {
    id: "layer-ui",
    label: "Web Dashboard",
    description: "The Next.js webapp (webapp/src/) providing an interactive browser-based dashboard for browsing and exploring installed Archon skills, agents, bundles, synapses, and pipelines.",
    color: "#ec4899",
    order: 7
  }
];

const nodeLayerMap = {};

function assign(id, layerId) {
  nodeLayerMap[id] = layerId;
}

for (const node of nodes) {
  const id = node.id;
  const path = node.path || id.replace(/^[^:]+:/, '');
  const type = node.type;

  if (type === 'pipeline') {
    assign(id, 'layer-infrastructure');
    continue;
  }

  if (type === 'document') {
    const p = id.replace('document:', '');
    if (p.startsWith('skills/') || p.startsWith('agents/') || p.startsWith('bundles/') || p.startsWith('synapses/') || p.startsWith('pipelines/')) {
      assign(id, 'layer-component-library');
    } else {
      assign(id, 'layer-tooling');
    }
    continue;
  }

  if (type === 'config') {
    const p = id.replace('config:', '');
    if (p.startsWith('.github/') || p.startsWith('.claude/')) {
      assign(id, 'layer-infrastructure');
    } else if (p.startsWith('skills/') || p.startsWith('agents/') || p.startsWith('bundles/') || p.startsWith('synapses/') || p.startsWith('pipelines/')) {
      assign(id, 'layer-component-library');
    } else if (p.startsWith('schemas/') || p === 'archon.yaml' || p === 'pyproject.toml' || p === 'catalog/catalog.yaml') {
      assign(id, 'layer-config');
    } else if (p.startsWith('webapp/')) {
      assign(id, 'layer-ui');
    } else if (p.startsWith('vscode-extension/')) {
      assign(id, 'layer-tooling');
    } else if (p.startsWith('servers/') || p.startsWith('file-ops-rs/')) {
      assign(id, 'layer-sdk-servers');
    } else if (p.startsWith('hooks/')) {
      assign(id, 'layer-runtime-hooks');
    } else {
      assign(id, 'layer-tooling');
    }
    continue;
  }

  // FILE, FUNCTION, CLASS
  const routePath = path || '';

  if (routePath.startsWith('src/archon/')) {
    assign(id, 'layer-core-engine');
  } else if (routePath.startsWith('hooks/')) {
    assign(id, 'layer-runtime-hooks');
  } else if (routePath.startsWith('webapp/')) {
    assign(id, 'layer-ui');
  } else if (routePath.startsWith('vscode-extension/')) {
    assign(id, 'layer-tooling');
  } else if (routePath.startsWith('sdk/')) {
    assign(id, 'layer-sdk-servers');
  } else if (routePath.startsWith('servers/') || routePath.startsWith('file-ops-rs/')) {
    assign(id, 'layer-sdk-servers');
  } else if (routePath.startsWith('scripts/')) {
    assign(id, 'layer-tooling');
  } else if (routePath.startsWith('tests/')) {
    assign(id, 'layer-tooling');
  } else if (routePath.startsWith('skills/') || routePath.startsWith('agents/') || routePath.startsWith('bundles/') || routePath.startsWith('synapses/') || routePath.startsWith('pipelines/')) {
    assign(id, 'layer-component-library');
  } else {
    assign(id, 'layer-tooling');
  }
}

const assigned = Object.keys(nodeLayerMap).length;
const unassigned = nodes.map(n => n.id).filter(id => !nodeLayerMap[id]);

console.log('Assigned:', assigned);
console.log('Unassigned:', unassigned.length);
if (unassigned.length > 0) {
  console.log('Unassigned sample:', unassigned.slice(0, 5));
}

const layerCounts = {};
for (const [id, layerId] of Object.entries(nodeLayerMap)) {
  layerCounts[layerId] = (layerCounts[layerId] || 0) + 1;
}

console.log('\nLayer counts:');
layers.forEach(l => console.log(l.order, l.label + ':', layerCounts[l.id] || 0));

const output = { layers, nodeLayerMap };
fs.writeFileSync('C:/Users/tahaa/Archon/.understand-anything/intermediate/architecture.json', JSON.stringify(output, null, 2), 'utf8');
console.log('\nWritten to architecture.json');

File: .understand-anything/.trash-1781812164/tmp/build-arch.py
import json

with open('C:/Users/tahaa/Archon/.understand-anything/intermediate/assembled-graph.json', 'r', encoding='utf-8') as f:
    data = json.load(f)
nodes = data['nodes']

def get_top(path):
    p = path.replace('\\', '/')
    parts = p.split('/')
    return parts[0] if len(parts) > 1 else '__root__'

def assign_layer(node_id, path, node_type):
    top = get_top(path)
    if top == 'hooks':
        return 'layer:runtime-hooks'
    if top == 'src':
        return 'layer:core-engine'
    if top == 'sdk':
        return 'layer:sdk-cli'
    if top in ('skills', 'agents', 'synapses', 'bundles', 'pipelines'):
        return 'layer:manifests'
    if top in ('webapp', 'vscode-extension', 'servers'):
        return 'layer:tooling'
    if top == 'file-ops-rs':
        return 'layer:core-engine'
    if top == 'scripts':
        return 'layer:sdk-cli'
    return 'layer:config-docs'

node_layer_map = {}
for n in nodes:
    nid = n['id']
    path = n.get('path', n.get('filePath', ''))
    ntype = n.get('type', '')
    node_layer_map[nid] = assign_layer(nid, path, ntype)

layers = [
    {
        'id': 'layer:config-docs',
        'label': 'Configuration & Documentation',
        'description': 'Root-level config files (pyproject.toml, archon.yaml), JSON/YAML schemas, CI/CD pipelines, documentation, prompts, and project-level settings that define and describe the Archon framework.',
        'color': '#6B7280',
        'order': 0
    },
    {
        'id': 'layer:manifests',
        'label': 'Manifest Components',
        'description': 'The declarative component library: skill YAML manifests and SKILL.md docs, agent definitions, synapse configurations, bundle groups, and execution pipeline definitions that form the installable Archon skill catalog.',
        'color': '#8B5CF6',
        'order': 1
    },
    {
        'id': 'layer:runtime-hooks',
        'label': 'Runtime Hooks',
        'description': 'Claude Code hook scripts (session_boot, prompt_router, guard_bash, guard_write, quality_bash, quality_write, completion_gate, agent_context) and shared utilities that inject contextual XML into every Claude Code prompt at runtime.',
        'color': '#F59E0B',
        'order': 2
    },
    {
        'id': 'layer:core-engine',
        'label': 'Core Engine',
        'description': 'The Python engine package (src/archon/) implementing the registry, pipeline state machine, synapse engine v2, policy engine, MCP schema synthesis, and the Rust file-ops library — the computational backbone that powers all Archon operations.',
        'color': '#EF4444',
        'order': 3
    },
    {
        'id': 'layer:sdk-cli',
        'label': 'SDK & CLI',
        'description': 'The public ArchonSDK (sdk/archon.py) exposing validate_* methods for external callers, the Typer CLI entrypoint, and scripts for automation tasks such as graph building and batch processing.',
        'color': '#10B981',
        'order': 4
    },
    {
        'id': 'layer:tooling',
        'label': 'Tooling & Interfaces',
        'description': 'The Next.js dashboard webapp for browsing skills and visualizing the knowledge graph, the VS Code extension for in-editor Archon integration, and the MCP server implementations that expose Archon capabilities to AI agents.',
        'color': '#3B82F6',
        'order': 5
    }
]

output = {
    'layers': layers,
    'nodeLayerMap': node_layer_map
}

out_path = 'C:/Users/tahaa/Archon/.understand-anything/intermediate/architecture.json'
with open(out_path, 'w', encoding='utf-8') as f:
    json.dump(output, f, indent=2)

print('Written to', out_path)
print('Total nodes:', len(node_layer_map))

from collections import Counter
counts = Counter(node_layer_map.values())
for layer in layers:
    lid = layer['id']
    lname = layer['label']
    print(f'  {lname}: {counts.get(lid, 0)} nodes')
